### DISCUSSION TITLE
How to set the frame rate of streaming?

### DISCUSSION THREAD

**Discussion by iqbal-h** [mpromonet/webrtc-streamer#564] [Q&A]
I am using webrtc-streamer as a server to stream on the client side. Let's say the server has a video playing in a video player. I run the webrtc streamer default without changing anything using the simple command ./webrtc-streamer. On the client side, I open the browser and start streaming the video by clicking on the video player tab on the top. The streaming works; however, I have noticed that the video is streamed at <=30 FPS. My question is, how can I increase the FPS to 60 such that the server captures and sends video at 60 FPS?

**Comment by mpromonet** [comment]
Hi,

This depends on backend stream framerate and also webrtc peerconnection.   
By now no media constraints are set but you can probably set constraints on framerate if supported (see https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getSupportedConstraints)

Best Regards,
Michel.

**Comment by iqbal-h** [comment]
Hi,

Thanks for the reply.
I did not understand "set constraints on framerate if supported".
Here is the scenario:
Server: Running webrtc-streamer that streams desktop as it is
Client: Running chromium where I give the IP of webrtc-server_IP:PortNumber in the browser and start streaming. 

The frame I observed on the server side is about 26 FPS.
So my question is how to set the streaming at 60 FPS for the server (webrtc-streamer)?

I am not sure how [MediaDevices/getSupportedConstraints](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getSupportedConstraints)) helps.
